### ISSUE TITLE
Closed downloading package manifest [package]

### ISSUE THREAD

**Issue by KodeurKubik** [oven-sh/bun#11471]
### What version of Bun is running?

1.1.10+5102a9443

### What platform is your computer?

Darwin 23.4.0 arm64 arm

### What steps can reproduce the bug?

It just happens to me, stable internet and quite fast. I have no idea how to reproduce on someone else's computer though.

`bun install [package]` gives output: 
```
bun add v1.1.10 (5102a944)
  🔍 express [6/6] 
error: ConnectionClosed downloading package manifest express
error: express@ failed to resolve
```

tested on express, vite, discord.js with bun i, install and add

To note: this happens when I try to :
- install in empty folder
- install when existing node_modules

### What is the expected behavior?

Expected, just install the packages.

**Comment by smaccoun** [comment]
This is has been happening to me for about 5 days now. I finally found a workaround, which was to simply downgrade to bun 
v1.0.9.

Workaround: `curl -fsSl https://bun.sh/install | bash -s "bun-v1.0.9"` 

Here's everything i tried fwiw

**SETUP**

1. Darwin 23.5.0 arm64 arm
2. Bun - 1.1.12+43f0913c3

**ATTEMPTS**

1. `rm -rf node_modules && rm bun.lockb && bun install`
2. `rm -rf ~/.bun && curl -fsSL https://bun.sh/install | bash`
3. bun upgrade --canary
4. Mixing all of the above

**Comment by Jarred-Sumner** [comment]
I think this is a duplicate of https://github.com/oven-sh/bun/issues/11423#issuecomment-2158485243

But I really wish there was a way to reproduce this or what the cause is.

**Comment by smaccoun** [comment]
Agreed it's a duplicate of that. I also wish there was a way to replicate it. I'll post some more info there

**Comment by psugihara** [comment]
FYI I found that I can repro this consistently when I turn on iCloud Private Relay. Turning it off always fixes.

<img width="486" alt="Screenshot 2024-08-22 at 9 15 20 AM" src="https://github.com/user-attachments/assets/1d4faeca-61f9-403a-9493-8555257083a3">

**Comment by psugihara** [comment]
Are others able to reproduce like this? To clarify I do really prefer having the relay on and this is the only problem I've ever had with it.

**Comment by calvinf** [comment]
> Are others able to reproduce like this? To clarify I do really prefer having the relay on and this is the only problem I've ever had with it.

Yes, able to reproduce consistently, I've had to mitigate by (temporarily) disabling private relay, too.

**Comment by smaccoun** [comment]
> Are others able to reproduce like this? To clarify I do really prefer having the relay on and this is the only problem I've ever had with it.

I haven't noticed it in a long time, but I'm quite sure it had to do with something of almost the exact same nature

**Comment by RiskyMH** [comment]
We've invested a ton of time into making Bun more reliable over the last year. I'm having a really hard time reproducing this. If you're still running into this issue, please let me know and I'll re-open this.

**Comment by calvinf** [comment]
@RiskyMH I appreciate all the improvements in Bun.

I still run into this issue with the latest Bun if I have on iCloud Private Relay. macOS Sequoia 15.2 (though I had it on previous macOS versions, too).

Given that it's a networking issue, it could be something else, but iCloud Private Relay is the one thing that I know I can turn off to get `bun install` working again.
